.github/workflows/main.yml

```diff
@@ -123,13 +157,15 @@ jobs:
 
   cli-test:
     name: CLI Test
+    needs: [test]
```

Comment:
I'd be in favour of duplicating the "Get Nightly Release URL" step from the `test` job in this job to prevent this job having to wait for the `test` job to complete.

Reply:
How about just a single job that gets the nightly URL (which should be very quick) and then both the `test` and `cli-test` job can depend on that? That would still let the tests run quickly but avoids duplicating the code and making spurious API calls.